src/main/java/org/terasology/logic/health/RegenAuthoritySystem.java

```diff
@@ -122,6 +122,10 @@ private void regenerate(long currentTime) {
         for (EntityRef entity : regenSortedByTime.values()) {
             RegenComponent regen = entity.getComponent(RegenComponent.class);
             HealthComponent health = entity.getComponent(HealthComponent.class);
+            if (regen == null || health == null) {
+                logger.debug("Entity '{}' without RegenComponent or HealthComponent scheduled for regeneration - skipping", entity);
+                continue;
+            }
             if (health != null && health.nextRegenTick < currentTime) {
```

Comment:
you don't need this null check after this.

Reply:
```suggestion
            if (health.nextRegenTick < currentTime) {
```